src/net/tcp.ts

```diff
@@ -46,20 +46,15 @@ export function endpointOfSocketInfo(info:freedom_TcpSocket.SocketInfo)
   return retval;
 }
 
-// Closes a socket, along with its freedomjs interface object.
-function destroyFreedomSocket_(socket:freedom_TcpSocket.Socket) : Promise<void> {
-  // Note:
-  //   freedom['core.tcpsocket'].close != freedom['core.tcpsocket']().close
-  // The former destroys the freedom interface & communication channels.
-  // The latter is a method on the constructed interface object that is on
-  // the instance of the freedomjs TCP socket API.
-  var destroy = () => {
-    freedom['core.tcpsocket'].close(socket);
-  };
-  return socket.close().then(destroy, (e:Error) => {
-    destroy();
-    return e;
-  });
+// Closes a socket's freedomjs interface object, i.e.:
+//   freedom['core.tcpsocket']().close
+//
+// This is different from:
+//   freedom['core.tcpsocket'].close
```

Comment:
Oh god, you're right -- and I looked extra hard at that. I shortened this, hopefully it's clearer now (and a lot shorter).